app/src/main/java/org/mozilla/focus/fragment/WebFragment.kt

```diff
@@ -0,0 +1,135 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.fragment
+
+import android.os.Bundle
+import android.text.TextUtils
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.webkit.WebView
+
+import org.mozilla.focus.R
+import org.mozilla.focus.locale.LocaleAwareFragment
+import org.mozilla.focus.locale.LocaleManager
+import org.mozilla.focus.session.Session
+import org.mozilla.focus.utils.AppConstants
+import org.mozilla.focus.web.IWebView
+
+import java.util.Locale
+
+/**
+ * Base implementation for fragments that use an IWebView instance. Based on Android's WebViewFragment.
+ */
+abstract class WebFragment : LocaleAwareFragment() {
+    private var webViewInstance: IWebView? = null
+    private var isWebViewAvailable: Boolean = false
+
+    abstract val session: Session?
+
+    /**
+     * Get the initial URL to load after the view has been created.
+     */
+    abstract val initialUrl: String?
+
+    /**
+     * Inflate a layout for this fragment. The layout needs to contain a view implementing IWebView
+     * with the id set to "webview".
+     */
+    abstract fun inflateLayout(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View
+
+    abstract fun createCallback(): IWebView.Callback
+
+    /**
+     * Adds ability to add methods to onCreateView without override because onCreateView is final.
+     */
+    abstract fun onCreateViewCalled()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val view = inflateLayout(inflater, container, savedInstanceState)
+
+        val actualWebView = view.findViewById<View>(R.id.webview)
+        webViewInstance = actualWebView as IWebView
+
+        isWebViewAvailable = true
+        webViewInstance!!.setCallback(createCallback())
+
+        val session = session
+
+        if (session != null) {
+            webViewInstance!!.setBlockingEnabled(session.isBlockingEnabled)
+            webViewInstance!!.setRequestDesktop(session.shouldRequestDesktopSite())
+        }
+
+        if (session == null || !session.hasWebViewState()) {
+            val url = initialUrl
+            if (!TextUtils.isEmpty(url)) {
+                webViewInstance!!.loadUrl(url)
+            }
+        } else if (!AppConstants.isGeckoBuild) {
+            webViewInstance!!.restoreWebViewState(session)
+        }
+
+        onCreateViewCalled()
+        return view
+    }
+
+    override fun applyLocale() {
+        val context = context
+        val localeManager = LocaleManager.getInstance()
+        if (!localeManager.isMirroringSystemLocale(context)) {
+            val currentLocale = localeManager.getCurrentLocale(context)
+            Locale.setDefault(currentLocale)
+            val resources = context!!.resources
```

Comment:
Instead of crashing when context isn't set, should we just use the current/default locale?